docs/data/sdks/sdk-architecture.md

```diff
@@ -0,0 +1,368 @@
+
+// Log levels defined:
+// None = 0,
+// Error = 1,
+// Warn = 2,
+// Verbose = 3,
+// Debug = 4,
+
+class MyLogger{
+  disable(): void {
+    console.log('Logger has been disabled');
+  }
+
+  enable(logLevel: amplitude.Types.LogLevel): void {
+    console.log('Logger has been enabled with level:', logLevel);
+  }
+
+  log(...args: any[]): void {
+    // Implementation for the log method
+    console.log(...args);
+  }
+
+  warn(...args: any[]): void {
+    // Implementation for the warn method
+    console.log(...args);
+  }
+
+  error(...args: any[]): void {
+    // Implementation for the error method
+    console.log(...args);
+  }
+
+  debug(...args: any[]): void {
+    // Implementation for the debug method
+    console.log(...args);
+  }
+}
+
+amplitude.init('API_KEY', 'OPTIONAL_USER_ID', {
+  loggerProvider: new MyLogger(),
+});
+```
+
+### Event 
+
+Events represent how users interact with your application. 
+
+`BaseEvent` represents a basic event with optional properties. You can also track other event properties in the field of `event_properties`
+
+```typescript
+// Next-gen Browser SDK example: track an event
+
+import * as amplitude from '@amplitude/analytics-browser';
+import {BaseEvent} from '@amplitude/analytics-types';
+
+const buttonClickedEvent: BaseEvent = {
+  event_type: 'Button Clicked',
+  event_properties: {
+    buttonColor: 'primary',
+  }
+}
+
+amplitude.track(buttonClickedEvent);
+```
+
+Some SDKs also allow you to track an event without creating one but just pass the event type as a string.
+
+```typescript
+// Next-gen Browser SDK example: track an event without creating one 
+
+import * as amplitude from '@amplitude/analytics-browser';
+
+// Track a basic event
+amplitude.track('Button Clicked');
+
+// Track events with optional properties
+const eventProperties = {
+  buttonColor: 'primary',
+};
+amplitude.track('Button Clicked', eventProperties);
+```
+
+### Plugin
+
+Plugins allow you to extend Amplitude SDK's behavior by, for example:
+
+- Use an Enrichment Plugin to modify event properties
+!!!example "Enrichment plugin examples"
+    
+    ???code-example "Remove PII (Personally Identifiable Information) (click to expand)"
+        
+        ```typescript
+        import * as amplitude from '@amplitude/analytics-browser';
+        import { EnrichmentPlugin, BrowserConfig, PluginType, Event } from '@amplitude/analytics-types';
+
+        class RemovePIIPlugin implements EnrichmentPlugin {
+        name = 'remove-PII-plugin';
+        type = PluginType.ENRICHMENT as any;
+
+        async setup(config: BrowserConfig): Promise<void> {
+            return undefined;
+        }
+
+        async execute(event: Event): Promise<Event> {
+            // remove PII here 
+
+            // return modified events with PII removed
+            return event
+        }
+        }
+
+        amplitude.init('API-KEY');
+        amplitude.add(new RemovePIIPlugin());
+        ```
```

Comment:
```suggestion
        ```typescript
        import * as amplitude from '@amplitude/analytics-browser';
        import { EnrichmentPlugin, BrowserConfig, PluginType, Event } from '@amplitude/analytics-types';

        class RemovePIIPlugin implements EnrichmentPlugin {
          name = 'remove-PII-plugin';
          type = PluginType.ENRICHMENT as any;
  
          async setup(config: BrowserConfig): Promise<void> {
              return undefined;
          }
  
          async execute(event: Event): Promise<Event> {
              // remove PII on the event
              if(event.user_properties['phone']) {
                delete event.user_properties['phone'];
                
                // set a new prop to mark this event as modified
                event.event_properties['pii-removed'] = true;
              }
  
              // return modified event with PII removed
              return event
          }
        }

        amplitude.init('API-KEY');
        amplitude.add(new RemovePIIPlugin());
        ```
```